setup.py

```diff
@@ -1,164 +1,89 @@
+        ]
 
-def _filter_tests(files):
-    testdir = join("pylint", "test")
-    return [f for f in files if testdir not in f]
 
-
-if easy_install_lib:
-
-    class easy_install(easy_install_lib.easy_install):
-        # override this since pip/easy_install attempt to byte compile
-        # test data files, some of them being syntactically wrong by design,
-        # and this scares the end-user
-        def byte_compile(self, files):
-            files = _filter_tests(files)
-            easy_install_lib.easy_install.byte_compile(self, files)
-
-
-def install(**kwargs):
-    """setup entry point"""
-    if USE_SETUPTOOLS:
-        if "--force-manifest" in sys.argv:
-            sys.argv.remove("--force-manifest")
-    packages = ["pylint"] + get_packages(join(base_dir, "pylint"), "pylint")
-    if USE_SETUPTOOLS:
-        if install_requires:
-            kwargs["install_requires"] = install_requires
-            kwargs["dependency_links"] = dependency_links
-        kwargs["entry_points"] = {
+def install():
+    setuptools.setup(
+        name="pylint",
+        version=__pkginfo__["__version__"],
+        license="GPL-2.0-or-later",
+        description="python code static checker",
+        long_description=long_description,
+        author="Python Code Quality Authority",
+        author_email="code-quality@python.org",
+        url="https://github.com/PyCQA/pylint",
+        project_urls={"What's New": "https://pylint.pycqa.org/en/latest/whatsnew/"},
+        classifiers=[
+            "Development Status :: 6 - Mature",
+            "Environment :: Console",
+            "Intended Audience :: Developers",
+            "License :: OSI Approved :: GNU General Public License (GPL)",
+            "Operating System :: OS Independent",
+            "Programming Language :: Python",
+            "Programming Language :: Python :: 3",
+            "Programming Language :: Python :: 3.6",
+            "Programming Language :: Python :: 3.7",
+            "Programming Language :: Python :: 3.8",
+            "Programming Language :: Python :: 3.9",
+            "Programming Language :: Python :: 3 :: Only",
+            "Programming Language :: Python :: Implementation :: CPython",
+            "Programming Language :: Python :: Implementation :: PyPy",
+            "Topic :: Software Development :: Debuggers",
+            "Topic :: Software Development :: Quality Assurance",
+            "Topic :: Software Development :: Testing",
+        ],
+        packages=setuptools.find_packages(),
+        python_requires="~=3.6",
+        setup_requires=[],
+        tests_require=test_requires,
+        install_requires=[
+            "astroid>=2.5.2,<2.7",
+            "isort>=4.2.5,<6",
+            "mccabe>=0.6,<0.7",
+            "toml>=0.7.1",
+        ],
+        extras_require={
+            ':sys_platform=="win32"': ["colorama"],
+            "docs": doc_extra_requires,
+            "dev": dev_extra_requires,
```

Comment:
I don't really see a need to add `dev` as an `extras_require` here. The usual way  should be to fork and clone the repo and then do
```bash
cd pylint
pip install -e .
pip install -r requirements_test.txt
```

Reply:
I like being able to install everything at once with ``pip3 install -e .[dev]``, would you mind if we keep it ?

Reply:
I understand your reasoning. However in that case we should probably loosen the test requirements inside `setup.py` and `setup.cfg`. `install_requires` and `extras_require` should usually **not** pin exact versions. From that viewpoint, I would still recommend installing the test requirements via `pip install -r requirements_test.txt`.

Reply:
I kept the old structure (I just removed astroid from test dependencies as it's a dependency already) and put it in the setup.cfg. Less files to keep track of I think. See https://github.com/PyCQA/pylint/pull/4294/commits/1291354f35730df4411249a7e544f6cf4dc118fe